### PR TITLE
Add Fix problem Task for Consult Beneficiaries

### LINKED ISSUE #116: Add Fix Problem Task When an Issue is to be Flagged in Beneficiary Consultation
With the addition of the beneficiary consultation task, we have a question on whether there is an issue at the service point that should qualify as `flag problem`. When a `Yes` is given, a problem is flagged. Previously whenever a problem was flagged for the inventories, there was a `Fix Problem` task generated that ensure that the issue flagged has been resolved. Since the `fix problem` task has been reinstated, we need to ensure that we also have the task generated in the above scenario.

### PR DESCRIPTION
**Pull request by hamza-vd**
Closes #116

### CODE REVIEW

File: opensrp-eusm/src/main/java/org/smartregister/eusm/viewholder/TaskRegisterViewHolder.java [hunk 1]
```diff
@@ -88,6 +89,8 @@ public void setProductImage(@Nullable TaskDetail taskDetail) {
                 this.productImageView.setImageDrawable(ResourcesCompat.getDrawable(context.getResources(), R.drawable.service_point_check_thumbnail, context.getTheme()));
             } else if (AppConstants.NonProductTasks.WAREHOUSE_CHECK.equalsIgnoreCase(taskDetail.getEntityName())) {
                 this.productImageView.setImageDrawable(ResourcesCompat.getDrawable(context.getResources(), R.drawable.service_point_check_thumbnail, context.getTheme()));
+            } else if (AppConstants.TaskCode.FIX_PROBLEM_CONSULT_BENEFICIARIES.equalsIgnoreCase(taskDetail.getTaskCode())) {
```
Comment (bennsimon) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/viewholder/TaskRegisterViewHolder.java:92]: Whats the value of `taskDetail.getEntityName()` here? If its the same as taskCode kindly use the entityName.

File: opensrp-eusm/src/main/java/org/smartregister/eusm/fragment/EusmTasksRegisterFragment.java [hunk 2]
```diff
@@ -130,6 +130,8 @@ public void onViewClicked(View view) {
                         presenter.startForm(structureDetail, taskDetail, getBeneficiaryConsultationForm());
                     } else if (AppConstants.NonProductTasks.WAREHOUSE_CHECK.equalsIgnoreCase(taskDetail.getEntityName().trim())) {
                         presenter.startForm(structureDetail, taskDetail, getWarehouseCheckForm());
+                    } else if (AppConstants.TaskCode.FIX_PROBLEM_CONSULT_BENEFICIARIES.equalsIgnoreCase(taskDetail.getTaskCode())) {
```
Comment (bennsimon) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/fragment/EusmTasksRegisterFragment.java:133]: Whats the value of `taskDetail.getEntityName()` here? If its the same as taskCode kindly use the entityName.

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppJsonFormUtils.java [hunk 3]
```diff
@@ -60,7 +60,8 @@ public JSONObject getFormObjectWithDetails(Context context, String formName, Str
             map.put(AppConstants.EventDetailKey.MISSION, PreferencesUtil.getInstance().getCurrentPlan());
             String entityId;
             if (AppConstants.JsonForm.RECORD_GPS_FORM.equals(formName) || AppConstants.JsonForm.SERVICE_POINT_CHECK_FORM.equals(formName)
-                    || AppConstants.JsonForm.BENEFICIARY_CONSULTATION_FORM.equals(formName) || AppConstants.JsonForm.WAREHOUSE_CHECK_FORM.equals(formName)) {
+                    || AppConstants.JsonForm.BENEFICIARY_CONSULTATION_FORM.equals(formName) || AppConstants.JsonForm.WAREHOUSE_CHECK_FORM.equals(formName)
+                    || (AppConstants.JsonForm.FIX_PROBLEM_FORM.equals(formName) && taskDetail.getTaskCode().equals(AppConstants.TaskCode.FIX_PROBLEM_CONSULT_BENEFICIARIES))) {
```
Comment (bennsimon) [inline opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppJsonFormUtils.java:64]:
```suggestion
                    || (AppConstants.JsonForm.FIX_PROBLEM_FORM.equals(formName) && AppConstants.TaskCode.FIX_PROBLEM_CONSULT_BENEFICIARIES.equals(taskDetail.getTaskCode())) {
```

Comment (bennsimon) [reply opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppJsonFormUtils.java:64]: switch the equals arg.

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppJsonFormUtils.java (hunk 3 above)
Comment (hamza-vd) [reply opensrp-eusm/src/main/java/org/smartregister/eusm/util/AppJsonFormUtils.java]: Done

**Review by bennsimon — APPROVED** [review]

**Pull request merged**